src/nrnpython/inithoc.cpp

```diff
@@ -124,37 +124,44 @@ void inithoc() {
 
 #if NRNMPI_DYNAMICLOAD
   nrnmpi_stubs();
-  // if nrnmpi_load succeeds (MPI available), pmes is nil.
-  pmes = nrnmpi_load(1);
-#endif //NRNMPI_DYNAMICLOAD
-
+  if (getenv("NEURON_INIT_MPI")) {
```

Comment:
@jorblancoa : could you try approach based on [this comment](https://github.com/neuronsimulator/nrn/issues/258#issuecomment-521397467)? i.e. in NRNMPI_DYNAMICLOAD case, we check if MPI_Init symbol is loaded into memory and then call nrnmpi_load.